### ISSUE TITLE
2.12.0 branch fails to extract

### ISSUE THREAD

**Issue by SethTisue** [scala/community-builds#401]
reported by @milessabin, and I reproduced it locally (with `git checkout -b 2.12.0 --track origin/2.12.0; version=2.12.0 ./run.sh`):

```
[info] Project cats--------------------: EXTRACTION FAILED (RuntimeException: The reference 7829269833d2b9ab0b508fa97...)
[info] Project lift-json---------------: EXTRACTION FAILED (RuntimeException: The reference ba4f884099c4011a01e5b5cc4...)
[info] Project monocle-----------------: EXTRACTION FAILED (RuntimeException: The reference 57701d0dc57b352ed52b4caca...)
[info] Project scalacheck--------------: EXTRACTION FAILED (RuntimeException: The reference 9b5a774ad8a0ea03fd5053522...)
[info] >>> The dbuild result is------------: EXTRACTION FAILED (failed: cats, lift-json, monocle, scalacheck)
```

the 2.12.0 branch isn't in active service on Jenkins anymore, but I think it does have some value as a stable target for testing compiler changes against (since the 2.12.x branch is constantly moving on and constantly breaking). so I'd like to at least understand what's going on here. perhaps it's an easy fix.

**Comment by SethTisue** [comment]
Those commits seem not to exist. It must have something to do with the method @adriaanm used in https://github.com/scala/community-builds/issues/338#issuecomment-253613276 to get frozen SHAs?

I don't know exactly which 2.12.x run he used, but it was October 13 that day, so maybe https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/791/consoleFull

yeah that must be the run, or close enough, all four SHAs are there:

```
[cats] Fetching http://github.com/SethTisue/cats.git
[cats] Commit: 7829269833d2b9ab0b508fa9797af63265a32eca
[lift-json] Fetching https://github.com/SethTisue/framework.git
[lift-json] Commit: ba4f884099c4011a01e5b5cc44ecb10961d88fc2
[monocle] Fetching https://github.com/julien-truffaut/Monocle.git
[monocle] Commit: 57701d0dc57b352ed52b4caca97e7c58cda0ff96
[scalacheck] Fetching https://github.com/rickynils/scalacheck.git
[scalacheck] Commit: 9b5a774ad8a0ea03fd5053522059bd39d79916a4
```

**Comment by SethTisue** [comment]
oh, I see now in Adriaan's commit comment for 53980196da6cd117127108f088b91683a66b50bf that 791 is in fact the right run.

the last commit before the freeze was 337cf8826c89157c1705a0782aeee888a1545ff, at which time we had:

```
project-refs.conf:  cats-uri:                     "https://github.com/SethTisue/cats.git#community-build-2.12"
project-refs.conf:  monocle-uri:                  "https://github.com/julien-truffaut/Monocle.git#scala-2.12-RC1-core"
project-refs.conf:  scalacheck-uri:               "https://github.com/rickynils/scalacheck.git#1.12.x"
project-refs.conf:  lift-json-uri:                "https://github.com/SethTisue/framework.git#community-build-2.12"
```

in the case of my own forks, I think I was treating those branches as WIP branches we didn't need to keep history on. presumably I later force-pushed them and the dangling commits have been garbage collected. Julien's Monocle branch was presumably similar.

the ScalaCheck one seems a bit mysterious, but whatevs, maybe someday Wikileaks will reveal what happened there, probably some coverup high up in the Swedish government, payoffs having to do with Dylan's Nobel, that kind of thing.

so, lesson learned: in the future, at the time we take one of these "snapshots" of the community build, it is necessary to take steps to make sure the commits in question get preserved, probably by forking all of the repos . I will leave remarks on this over at https://github.com/scala/community-builds/issues/338

in the short term, I'll try and find usable SHAs for these four projects to make the 2.12.0 branch usable again.

**Comment by SethTisue** [comment]
fixed in #406
